openquake/hazardlib/gsim/base.py

```diff
@@ -912,20 +912,46 @@ class CoeffsTable(object):
     Traceback (most recent call last):
         ...
     KeyError: SA(period=0.01, damping=5)
+
+    It is also possible to instantiate a table from a tuple of dictionaries,
+    corresponding to the SA coefficients and non-SA coefficients:
+
+    >>> coeffs = {SA(0.1): {"a": 1.0, "b": 2.0},
+    ...           SA(1.0): {"a": 3.0, "b": 4.0},
+    ...           PGA(): {"a": 0.1, "b": 1.0},
+    ...           PGV(): {"a": 0.5, "b": 10.0}}
+    >>> ct = CoeffsTable(sa_damping=5, table=coeffs)
     """
     def __init__(self, **kwargs):
         if 'table' not in kwargs:
             raise TypeError('CoeffsTable requires "table" kwarg')
-        table = kwargs.pop('table').strip().splitlines()
+        table = kwargs.pop('table')
+        self.sa_coeffs = {}
+        self.non_sa_coeffs = {}
         sa_damping = kwargs.pop('sa_damping', None)
         if kwargs:
             raise TypeError('CoeffsTable got unexpected kwargs: %r' % kwargs)
+        if isinstance(table, str):
+            self._setup_table_from_str(table, sa_damping)
```

Comment:
Did you run the tests both with Python 2 and Python 3? I wonder if the str/bytes change may cause trouble here.